mlflow/types/utils.py

```diff
@@ -93,15 +93,16 @@ def _infer_schema(data: Any) -> Schema:
     if any([t in (DataType.integer, DataType.long) for t in schema.column_types()]):
         warnings.warn(
             "Hint: Inferred schema contains integer column(s). Integer columns in "
-            "Python can  not represent missing values. If your input data contains"
+            "Python can  not represent missing values. If your input data contains "
             "missing values at inference time, it will be encoded as floats and will "
             "cause a schema enforcement error. The best way to avoid this problem is "
             "to infer the model schema based on a realistic data sample (training "
             "dataset) that includes missing values. Alternatively, you can declare "
             "integer columns as doubles (float64) whenever these columns may have "
             "missing values. See `Handling Integers With Missing Values "
             "<https://www.mlflow.org/docs/latest/models.html#"
-            "handling-integers-with-missing-values>`_ for more details."
+            "handling-integers-with-missing-values>`_ for more details.",
+            stacklevel=2,
```

Comment:
`stacklevel=2` refers to the line that triggered the warning, rather than the line on which the `warnings.warn` call is placed.